build 2 row view for cells

Deardrops opened the pull request:
TODO:

- [x] 处理长列表优化
- [x] lrig卡组显示有bug

Deardrops committed:
2 row view by media query

Deardrops committed:
add cellcontainer component with long list optimization

webxoss commented on src/components/CellContainer.vue:
```diff
@@ -0,0 +1,117 @@
+  }),
+  computed: {
+    isTwoColunm() {
+      return window.screen.width > 768
```
不是响应式的。
把宽度放到 store 里， window resize 的时候变异它。

webxoss commented on src/components/CellContainer.vue:
```diff
@@ -0,0 +1,117 @@
+        return Math.ceil(this.cards.length / 2)
+      } else {
+        return this.cards.length
+      }
```
用三元表达式吧。
```
return longLongLongCondition
  ? longLongValue
  : longLongValue
```
这样的缩进风格

webxoss commented on src/components/CellContainer.vue:
```diff
@@ -0,0 +1,117 @@
+}
+.col {
+  width: 100%;
+  @media (width > 768px) {
```
应该是`>=` 吧？
iPad 的竖屏就是 768px

Deardrops committed:
responsive behavior of one/two colunm view

1. 768 < width < 1024, use 1 colunm view with 24px font size
2. width > 1024, use 2 colunm view with 20px font size
3. width > 1280, use 2 colunm view with 24px font size

store window's width in `vuex`, update when resize window
prop `longListOpimizationEnable` arg for using it in `cellContainer` component

webxoss commented on src/store.js:
```diff
@@ -207,4 +211,8 @@ store.watch((state, getters) => {
 })
 
+window.onresize = function () {
+  store.commit('changeWindowWidth', window.document.body.clientWidth)
```
`window.innerWidth`吧
http://stackoverflow.com/questions/6942785/window-innerwidth-vs-document-documentelement-clientwidth

webxoss commented on src/store.js:
```diff
@@ -207,4 +211,8 @@ store.watch((state, getters) => {
   localStorage.setItem(deckName, JSON.stringify(file))
 })
 
+window.onresize = function () {
```
`addEventListener`

webxoss commented on src/pages/Deck.vue:
```diff
@@ -123,7 +123,7 @@ export default {
     },
     updateDeckHeader() {
-      let $lrigDeck = this.$refs.lrigDeck
+      let $lrigDeck = this.$refs.lrigDeck.$el // TODO: check null
```
这里不要 todo 了。。
直接 do it

webxoss commented on src/app.vue:
```diff
@@ -5,6 +5,7 @@ export default {
     let deckFileNames = localStorage.getItem('deck_filenames')
     let fileNames = JSON.parse(deckFileNames)
 
+    this.$store.commit('changeWindowWidth', window.screen.width)
```
不要用 screen

Deardrops committed:
review

webxoss commented on src/components/CellContainer.vue:
```diff
@@ -0,0 +1,125 @@
+    },
+  },
+  mounted() {
+    if (this.longListOpimizationEnabled) {
```
这里吹毛求疵一点的话，应该 watch ，响应这个属性的变化，
嘛，这样也没问题啦

webxoss commented on src/pages/Deck.vue:
```diff
@@ -123,11 +123,14 @@ export default {
-      this.scrolledToTop = window.scrollY <= 0
-      this.request = requestIdleCallback(this.updateDeckHeader)
+      let lrigDeck = this.$refs.lrigDeck
+      if (lrigDeck) {
```
提前结束吧
```
if (!lrigDeck) {
  return
}
```

webxoss commented on src/store.js:
```diff
@@ -211,8 +211,8 @@ store.watch((state, getters) => {
-}
+window.addEventListener('resize', () => {
+  store.commit('changeWindowWidth', window.innerWidth)
+})
```
加上第三个参数吧

webxoss reviewed (commented):
我本地跑跑看

webxoss reviewed (approved):
可以合了，我提那些可改可不改。

2列时的详情和画廊视图也得稍微设计一下，另提pr吧。
